### PR TITLE
app: fix peer count bn query to occur every min instead of every 10s

### PR DESCRIPTION
**Pull request by xenowits**
Fixes peer count beacon node query to occur every min instead of every 10s. 

Also, fixes it to query only once since we query this both in `startReadyzChecker()` and `beaconNodeMetrics()` and set the `beaconNodeVersionGauge` in `startReadyzChecker` instead.

category: misc 
ticket: none

### CODE REVIEW

File: app/monitoringapi.go [hunk 1]
```diff
@@ -143,7 +147,7 @@ func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2wrap.C
 				} else if syncing {
 					err = errReadyBeaconNodeSyncing
 					readyzGauge.Set(readyzBeaconNodeSyncing)
-				} else if bnPeerCount == 0 && bnErr == nil {
+				} else if bnPeerCount == 0 {
```
Comment (corverroos) [inline app/monitoringapi.go:150]: this will trigger if `NodePeerCount` never ran or if it failed previously

File: app/monitoringapi.go [hunk 2]
```diff
@@ -115,26 +120,38 @@ func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2wrap.C
 			prevPKs[pubkey] = true
 		}
 
+		// beaconNodePeerCount queries beacon node peer count and sets the peer count gauge if err is nil.
+		beaconNodePeerCount := func() {
+			bnPeerCount, bnPeerCountErr = eth2Cl.NodePeerCount(ctx)
+			if bnPeerCountErr != nil {
+				log.Error(ctx, "Failed to get beacon node peer count", bnPeerCountErr)
```
Comment (corverroos) [inline app/monitoringapi.go:127]: suggest warn

File: app/monitoringapi.go [hunk 3]
```diff
@@ -115,26 +120,38 @@ func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2wrap.C
...
+
+		onStartup := make(chan struct{}, 1)
+		onStartup <- struct{}{}
+
 		for {
 			select {
 			case <-ctx.Done():
 				return
+			case <-onStartup:
+				beaconNodePeerCount()
```
Comment (corverroos) [inline app/monitoringapi.go:141]: lets remove this onstartup part, it isn't important

File: app/monitoringapi.go [hunk 4]
```diff
@@ -106,7 +106,12 @@ func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2wrap.C
 	)
 	go func() {
 		ticker := clock.NewTicker(10 * time.Second)
+		peerCountTicker := clock.NewTicker(1 * time.Minute)
 		epochTicker := clock.NewTicker(32 * 12 * time.Second) // 32 slots * 12 second slot time
+		var (
+			bnPeerCount    int   // Beacon node peer count value which is queried on startup & every minute
```
Comment (corverroos) [inline app/monitoringapi.go:112]: slightly simpler, lets use a `bnPeerCount *bool`,  then below use `if bnPeerCount != nil && *bnPeerCount == 0`

Comment (xenowits) [reply app/monitoringapi.go]: i suppose you mean `bnPeerCount *int`

File: app/monitoringapi.go [hunk 5]
```diff
@@ -115,6 +117,20 @@ func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2wrap.C
...
+		beaconNodePeerCount := func() {
+			peerCount, err := eth2Cl.NodePeerCount(ctx)
+			if err != nil {
+				log.Warn(ctx, "Failed to get beacon node peer count", err)
+				return
+			}
+			if bnPeerCount == nil {
+				bnPeerCount = new(int)
+			}
+			*bnPeerCount = peerCount
```
Comment (corverroos) [inline app/monitoringapi.go:130]: replace with: `bnPeerCount = &peerCount`

**Review by corverroos — APPROVED** [review]

**Pull request merged**